staging/src/k8s.io/apiserver/pkg/server/lifecycle_signals.go

```diff
@@ -144,6 +148,7 @@ func newLifecycleSignals() lifecycleSignals {
 		ShutdownInitiated:          newNamedChannelWrapper("ShutdownInitiated"),
 		AfterShutdownDelayDuration: newNamedChannelWrapper("AfterShutdownDelayDuration"),
 		InFlightRequestsDrained:    newNamedChannelWrapper("InFlightRequestsDrained"),
+		PreShutdownHooksCompleted:  newNamedChannelWrapper("PreShutdownHooksCompleted"),
```

Comment:
promoted a new lifecycle signal for when preshutdown hooks have completed

Reply:
This looks great